compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2249,39 +2249,29 @@ static void arraycopyForShortConstArrayWithoutDirection(TR::Node* node, TR::Regi
 
 TR::Register *OMR::X86::TreeEvaluator::arraycopyEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   if (node->isReferenceArrayCopy() && !node->isNoArrayStoreCheckArrayCopy())
-      {
-      return TR::TreeEvaluator::VMarrayStoreCheckArrayCopyEvaluator(node, cg);
-      }
-   // There are two cases.
-   // In the first case we know that a simple memmove or memcpy operation can
-   // be done. For this case there are 3 children:
-   //    1) The byte source pointer
-   //    2) The byte destination pointer
-   //    3) The byte count
+   // There are two variants of TR::arraycopy: one has 5 children, the other has 3 children. Details can be found from
+   // compiler/il/ILOpCodeProperties.hpp
    //
-   // In the second case we must generate run-time tests to see if a simple
-   // byte copy can be done or if an element-by-element copy is needed.
-   // For this case there are 5 children:
-   //    1) The original source object reference
-   //    2) The original destination object reference
-   //    3) The byte source pointer
-   //    4) The byte destination pointer
-   //    5) The byte count
+   // In most, if not all, cases, the 5-child variant requires language specific semantics, which OMR is not aware of.
+   // The fact that a 5-child arraycopy is generated indicates at least one of the first two children must needed when
+   // performing the copy; otherwise a 3-child arraycopy should be generated instead. Interpreting the meanings of the
+   // first two children definitely requires language specific semantics. For example, the first two children may be for
+   // dealing with arraycopy where Garbage Collector may need be notified about the copy or something to that affect.
```

Comment:
with _an_ arraycopy where _the_ Garbage Collector may need _to_ be notified

Reply:
Updated.